modules/php/Cards/Rules/RuleSwapPlaysForDraws.php

```diff
@@ -9,22 +9,50 @@ public function __construct($cardId, $uniqueId)
   {
     parent::__construct($cardId, $uniqueId);
 
-    $canBeUsedByPlayer = true;
-
     $this->name = clienttranslate("Swap Plays for Draws");
     $this->subtitle = clienttranslate("Takes Instant Effect");
     $this->description = clienttranslate(
       "During your yurn, you may decide to play no more cards and instead draw as many cards as you have plays remaining. If Play All, draw as many cards as you hold."
     );
   }
 
+  public function canBeUsedInPlayerTurn($player_id)
+  {
+    $drawCount = $this->countSwapPlaysForDraws($player_id);
+    return $drawCount > 0;
+  }
+
   public function immediateEffectOnPlay($player)
   {
-    // @TODO
+    // nothing
   }
 
   public function immediateEffectOnDiscard($player)
   {
     // nothing
   }
+
+  private function countSwapPlaysForDraws($player_id)
+  {
+    $game = Utils::getGame();
+    // calculate how many cards player should still play
+    $drawCount = Utils::calculateCardsLeftToPlayFor($player_id);
```

Comment:
This function is simply calling another function. Maybe we can remove it for readability.

Reply:
It was a separate function originally, then refactored to make sure we only had the calculate logic in 1 central place.
So yes, can be cleaned.